rpc/grpc/client/privileged/pruning_service.go

```diff
@@ -0,0 +1,97 @@
+package privileged
+
+import (
+	"context"
+
+	"github.com/cosmos/gogoproto/grpc"
+
+	v1 "github.com/cometbft/cometbft/proto/tendermint/services/pruning/v1"
+)
+
+// RetainHeights provides information on which block height limits have been
+// set for block information to be retained by the ABCI application and the
+// pruning service.
+type RetainHeights struct {
+	App            uint64
+	PruningService uint64
+}
+
+type PruningServiceClient interface {
+	SetBlockRetainHeight(ctx context.Context, height uint64) error
+	GetBlockRetainHeight(ctx context.Context) (RetainHeights, error)
```

Comment:
Should we not have two separate functions here:

```suggestion
	GetBlockRetainHeight(ctx context.Context) (uint64, error)
	GetRetainHeights(ctx context.Context) (RetainHeights, error)
```

Reply:
Ah yes, we want to return both here. Nvm!